src/pybind/CMakeLists.txt

```diff
@@ -1,62 +1,74 @@
   set(module_source_files "${module_source_files}" usb.cpp)
 endif()
 
 # find_library(MetavisionSDK NAMES metavision_sdk_core REQUIRED)
 
-# Thanks to https://github.com/scikit-build/scikit-build-sample-projects/blob/master/projects/hello-pybind11/CMakeLists.txt
-find_package(PythonExtensions REQUIRED)
-find_package(pybind11 CONFIG)
-if (NOT pybind11_FOUND)
-  FetchContent_Declare(
-    pybind11
-    URL https://github.com/pybind/pybind11/archive/refs/tags/v2.9.2.tar.gz
-    URL_HASH SHA256=6bd528c4dbe2276635dc787b6b1f2e5316cf6b49ee3e150264e455a0d68d19c1
-  )
-  FetchContent_MakeAvailable(pybind11)
+# Thanks to https://github.com/wjakob/nanobind_example/blob/master/CMakeLists.txt
+# Create CMake targets for all Python components needed by nanobind
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.26)
+  find_package(Python 3.8 COMPONENTS Interpreter Development.Module Development.SABIModule REQUIRED)
+else()
+  find_package(Python 3.8 COMPONENTS Interpreter Development.Module REQUIRED)
 endif()
 
-set(PYTHON_MODULE_NAME aestream_ext)
-pybind11_add_module(${PYTHON_MODULE_NAME} MODULE
-      module.cpp 
-      udp.cpp 
-      udp_client.cpp 
-      udp_client.hpp 
-      iterator.cpp
-      file.hpp
-      file.cpp
-      # convert.cpp 
-      # convert.hpp 
-      tensor_buffer.hpp 
-      tensor_buffer.cpp
-      tensor_iterator.hpp
-      tensor_iterator.cpp
-      ${module_source_files})
-
-install(IMPORTED_RUNTIME_ARTIFACTS aer aestream_file aestream_input aestream_output DESTINATION aestream)
-
-target_compile_definitions(${PYTHON_MODULE_NAME} PUBLIC ${module_compile_definitions})
-target_compile_features(${PYTHON_MODULE_NAME} PUBLIC cxx_std_20)
-target_include_directories(${PYTHON_MODULE_NAME} PRIVATE "${TORCH_INCLUDE_DIRS}" "${CMAKE_INSTALL_LIBDIR}" "${PYTHON_SITE_PACKAGES_DIR}/aestream")
-target_link_libraries(${PYTHON_MODULE_NAME} PRIVATE "${TORCH_LIBRARIES}" "${TORCH_PYTHON_LIBRARY}" aer aestream_file aestream_input aestream_output)
-set_target_properties(${PYTHON_MODULE_NAME} PROPERTIES 
+# Run `nanobind.cmake_dir()` from Python to detect where nanobind is installed
+execute_process(
+  COMMAND "${Python_EXECUTABLE}" -c "import nanobind; print(nanobind.cmake_dir())"
+  OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE NB_DIR)
+list(APPEND CMAKE_PREFIX_PATH "${NB_DIR}")
+
+find_package(nanobind CONFIG REQUIRED)
+
+nanobind_add_module(
+  aestream_ext 
+  NB_STATIC STABLE_ABI LTO 
+  NOMINSIZE # Remove minimization to prevent -Os flags to propagate to nvcc
+
+  "${module_source_files}" # Include e. g. camera vendors 
+  module.cpp 
+  udp.cpp 
+  udp_client.cpp 
+  udp_client.hpp
+  # iterator.cpp
+  file.hpp
+  file.cpp
+  tensor_buffer.hpp
+  tensor_buffer.cpp
+  tensor_iterator.hpp
+  tensor_iterator.cpp
+)
+install(TARGETS aer aestream_file aestream_input aestream_output LIBRARY DESTINATION aestream)
+
+set_target_properties(aestream_ext PROPERTIES 
+  BUILD_RPATH ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES} # For Apple
```

Comment:
Previously when I was building, this was causing me issues so had to comment out. Need to double check

Reply:
`${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}` does not resolve to anything if cuda is not installed.
This means that there is the wrong number of arguments for `set_target_properties()` in this case

Reply:
Excellent point! And thank you for addressing that @cameron-git! :partying_face: